MobiFlight/Joysticks/VKB/VKBDevice.cs

```diff
@@ -0,0 +1,192 @@
+﻿using HidSharp;
+using HidSharp.Reports;
+using HidSharp.Reports.Input;
+using System;
+using System.Collections.Generic;
+
+namespace MobiFlight.Joysticks.VKB
+{
+    internal class VKBDevice : Joystick
+    {
+        private HidStream Stream;
+        private readonly HidDevice Device;
+        private readonly JoystickDefinition Definition;
+        private readonly VKBLedContainer Lights = new VKBLedContainer();
+        private HidDeviceInputReceiver InputReceiver;
+        private readonly byte[] InputReportBuffer = new byte[64];
+        private readonly SortedList<byte, VKBEncoder> Encoders = new SortedList<byte, VKBEncoder>();
+        int lastSeqNo = -1;
+
+        public VKBDevice(SharpDX.DirectInput.Joystick joystick, JoystickDefinition definition) : base(joystick, definition)
+        {
+            Definition = definition;
+            if (Device == null)
+            {
+                Device = GetMatchingHidDevice(joystick);
+                if (Device == null) return;
+            }
+        }
+        public override void Connect(IntPtr handle)
+        {
+            base.Connect(handle);
+            if (Stream == null)
+            {
+                Stream = Device.Open();
+                Stream.ReadTimeout = System.Threading.Timeout.Infinite;
+            }
+
+            if (InputReceiver == null)
+            {
+                InputReceiver = new ReportDescriptor(VKBHidReport.Descriptor).CreateHidDeviceInputReceiver();
+                // We use our own descriptor since the one we get from Windows/HIDSharp has been altered and encoder data is missing
+                InputReceiver.Received += OnHidReportReceived;
+                InputReceiver.Start(Stream);
+            }
+        }
+        protected override void SendData(byte[] data)
```

Comment:
Minor nit but there should be blank lines between each function definition to stay consistent with our style in other files.